analyzer/runtime/extract.go

```diff
@@ -578,6 +582,65 @@ func sumGasUsed(events []*EventData) (sum uint64, foundGasUsedEvent bool) {
 	return
 }
 
+// Attempts to extract the human-readable error message from
+// the raw error returned by the node.
+//
+// In Emerald and older Sapphire versions, transaction revert
+// reasons were decoded by the runtime and returned as
+// human-readable strings, e.g.
+// - "reverted: Incorrect premium amount."
+// - "execution failed: out of gas"
+//
+// More recent versions of Sapphire no longer decode the error
+// message and instead use the following error format:
+// - "reverted: " || base64(abiEncode(error))
+//
+// We first check to see if the error message is a plaintext error string,
+// in which case we simply sanitize and return the message.
+// Otherwise, we optimistically try to decode the error as the prevailing
+// Error(string) type and return the error message if successful.
+//
+// Note that the error type can be any data type specified in the abi.
+// We may not have the abi available now, so in those cases the
+// abi analyzer will extract the error message once the contract
+// abi is available.
+func tryParseErrorMessage(msg string) *string {
+	// Decode revert reason, format is as follows:
+	// "reverted: " || base64(abiEncode(error))
+	if !strings.HasPrefix(msg, "reverted: ") {
```

Comment:
Note that one could also check the error code. The reversion error has a specific error code and anything else should not be treated as a reversion:
```rust
    #[error("reverted: {0}")]
    #[sdk_error(code = 8)]
    Reverted(String),
```
So if module is `evm` and code is `8` the error is a reversion. Otherwise it shouldn't be parsed as one.

Reply:
got it, updated!